### PR TITLE
v9 aws fixes

### PR DESCRIPTION
**Pull request by vjeffrey**
fixes https://github.com/mondoohq/cnquery/issues/1640

### CODE REVIEW

File: providers/aws/resources/discovery_conversion.go [hunk 1]
```diff
@@ -195,8 +195,9 @@ func accountAsset(conn *connection.AwsConnection, awsAccount *mqlAwsAccount) *in
 		alias = aliases.Data[0].(string)
 	}
 	name := AssembleIntegrationName(alias, awsAccount.Id.Data)
+	justId := strings.TrimPrefix(awsAccount.Id.Data, "aws.account/")
```
Comment (czunker) [inline providers/aws/resources/discovery_conversion.go]: nit: Could we change this to `id` and the line below to `platformId` or something alike?

`justId` reads strangely to me.

Comment (czunker) [reply providers/aws/resources/discovery_conversion.go]: Or replace `justId` with `awsAccountId`.

File: providers/aws/resources/aws_ecs.go [hunk 2]
```diff
@@ -81,9 +81,9 @@ func (a *mqlAwsEcs) containerInstances() ([]interface{}, error) {
 	return containerInstances, nil
 }
 
-// func (a *mqlAwsEcsInstance) GetEc2Instance() ([]interface{}, error) {
-// 	return nil, nil
-// }
+func (a *mqlAwsEcsInstance) ec2Instance() (*mqlAwsEc2Instance, error) {
+	return nil, nil
```
Comment (czunker) [inline providers/aws/resources/aws_ecs.go:85]: Is this intended?

**Comment by czunker** [comment]
Looks like the aws provider still has some problems on my machine. I tried different queries:
```
cnquery run aws --discover instances -c 'asset{ platform name ids }'                                                                                             
→ loaded configuration from /etc/opt/mondoo/mondoo.yml using source default
! no public ip address found asset=ebs-scanner
! no public ip address found asset=ebs-scanner
! no public ip address found asset=ebs-scanner
! no public ip address found asset=ebs-scanner
! no public ip address found asset=ebs-scanner
! no public ip address found asset=ebs-scanner
! no public ip address found asset=ebs-scanner
! no public ip address found asset=ebs-scanner
! no public ip address found asset=ebs-scanner
! no public ip address found asset=ebs-scanner
! no public ip address found asset=ivan-scanner
! no public ip address found asset=k8s-operator01
! no public ip address found asset=k8s-operator02
! no public ip address found asset=k8s-operator03
! no public ip address found asset=amazonlinux2-for-ebs-volume-scan
FTL failed to run query error="asset has no connections, can't detect provider"
```
This one looks a bit better:
```
cnquery run aws --discover ssm-instances -c 'asset{ platform name ids }'                                                                                             
→ loaded configuration from /etc/opt/mondoo/mondoo.yml using source default
! cannot use ssm session credentials asset=mi-054b97883ba2375c2 id=mi-054b97883ba2375c2
asset: {
  platform: ""
  ids: [
    0: "//platformid.api.mondoo.app/runtime/aws/ec2/v1/accounts/aws.account/921877552404/regions/us-east-1/instances/mi-054b97883ba2375c2"
  ]
  name: "mi-054b97883ba2375c2"
}
```
But there the platformId has the wrong `aws.account/ID` in it. Strange. That should be fixed with this PR.

What I did to test this:
- checkout branch
- build was provider
- install providers
- run queries

**Comment by vjeffrey** [comment]
ohh, nice catch on the id!

for the instances, ill take a look at the diff between v8 and v9 -- when there's no public ip we can't give it an ssh connection - ill figure that out

**Comment by vjeffrey** [comment]
ok, id and discovery stuff should be fixed now

**Comment by vjeffrey** [comment]
```
[8/09/23 00:58:32] ❯ go run apps/cnquery/cnquery.go shell aws --profile home --discover instances
! using builtin provider for aws
→ loaded configuration from /Users/vj/.config/mondoo/mondoo.yml using source default
→ selected asset asset=test selection=0
→ connected to Amazon Linux 2023
  ___ _ __   __ _ _   _  ___ _ __ _   _
 / __| '_ \ / _` | | | |/ _ \ '__| | | |
| (__| | | | (_| | |_| |  __/ |  | |_| |
 \___|_| |_|\__, |\__,_|\___|_|   \__, |
  mondoo™      |_|                |___/  interactive shell

cnquery>
```

**Comment by vjeffrey** [comment]
oh but also, the aws cleanup that was merged yesterday (the one for the init empty structs to nil), seems to have broken `aws.ec2.instances { * }`, and a few other resources. ill work on that separately

**Comment by czunker** [comment]
ssm instance ID is looking good now:
```
cnquery run aws --discover ssm-instances -c 'asset{ platform name ids }' 
→ loaded configuration from /etc/opt/mondoo/mondoo.yml using source default
! cannot use ssm session credentials for connection asset= id=mi-054b97883ba2375c2
asset: {
  platform: "aws-ssm-instance"
  ids: [
    0: "//platformid.api.mondoo.app/runtime/aws/ssm/v1/accounts/921877552404/regions/us-east-1/instance/mi-054b97883ba2375c2"
    1: "arn:aws:ssm:us-east-1:921877552404:instance/mi-054b97883ba2375c2"
  ]
  name: "mi-054b97883ba2375c2"
}
```

**Comment by czunker** [comment]
The ed2 instance scan is also looking better:
```
...
asset: {
  platform: "aws-ec2-instance"
  ids: [
    0: "//platformid.api.mondoo.app/runtime/aws/ec2/v1/accounts/921877552404/regions/us-east-1/instance/i-09ee778fbcc3bd237"
    1: "arn:aws:ec2:us-east-1:921877552404:instance/i-09ee778fbcc3bd237"
  ]
  name: "ebs-scanner"
}
asset: {
  platform: "aws-ec2-instance"
  ids: [
    0: "//platformid.api.mondoo.app/runtime/aws/ec2/v1/accounts/921877552404/regions/us-east-1/instance/i-046b7cd9532b7afba"
    1: "arn:aws:ec2:us-east-1:921877552404:instance/i-046b7cd9532b7afba"
  ]
  name: "ebs-scanner"
}
...
```

Only thing is, I receive this error for an instance with a public IP, where I don't have access:
```
FTL failed to run query error="rpc error: code = Unknown desc = dial tcp 13.57.16.229:22: connect: connection timed out"
```
Not sure whether this is intended or not.

**Review by czunker — APPROVED** [review]
Thanks @vjeffrey

**Comment by vjeffrey** [comment]
"Only thing is, I receive this error for an instance with a public IP, where I don't have access" <- none of the instances in our dev aws account are public reachable right now, so ya that's expected

File: providers/aws/resources/aws_ecs.go (hunk 2 above)
Comment (vjeffrey) [reply providers/aws/resources/aws_ecs.go:85]: ya, it's an optional one

File: providers/aws/config/config.go [hunk 3]
```diff
@@ -13,7 +13,7 @@ var Config = plugin.Provider{
 	Name:            "aws",
 	ID:              "go.mondoo.com/cnquery/providers/aws",
 	Version:         "9.0.0",
-	ConnectionTypes: []string{provider.ConnectionType},
+	ConnectionTypes: []string{provider.DefaultConnectionType, provider.SshConnectionType, provider.RegistryImageConnectionType},
```
Comment (czunker) [inline providers/aws/config/config.go:16]: Just came across this. This seems wrong. The handled types should be in the provider which can actually take care of these.

I discovered k8s container images and was then redirected to the aws provider, although the os provider should handle the container images.

Comment (vjeffrey) [reply providers/aws/config/config.go:16]: oh, good catch! thanks, ill adjust

**Pull request merged**